### ISSUE TITLE
Solving NODE with implicit adjoint & steady state fails cause event occurred

### ISSUE THREAD

**Issue by daveekr** [patrick-kidger/diffrax#520]
Hello,

I am trying to solve a neural ODE using the `diffrax.ImplicitAdjoint()`, adaptive step sizing and the steady state event as proposed in the documentation. However, I get this error message which I do not fully understand. 

> equinox.EquinoxRuntimeError: Terminating differential equation solve because an event occurred.

I have `t0 = 0` and `t1 = jnp.inf` with this event
```python
cond_fn = diffrax.steady_state_event(rtol=1e-4, atol=1e-6)
event = diffrax.Event(cond_fn)
```

From my understanding: I actually want the solution at the steady state (`diffrax.SaveAt(t1=True)`), so I would expect that the event is fired once the steady state is reached and consequently the differential equation solve is being stopped. Otherwise the solver would solve forever, since `t1 = jnp.inf`, unless this was not capped by `max_steps`. Am I correct?

Any help is much appreciated, thanks!

**Comment by patrick-kidger** [comment]
Can you provide a MWE demonstrating the issue?

**Comment by daveekr** [comment]
Hi Patrick,

thanks for the quick response. Unfortunately, I cannot provide a MWE for this issue. The error occurs during training. I tried to isolate the failing conditions however, then the solver fails with a different error (`max_steps reached`).

However, I found out that I can set `throw=False` and evaluate the returned `diffrax.Solution` objects. Handling the `diffrax.RESULTS` enumeration is not that trivial which then led me to issue #442. Where you responded:

> (Also heads-up that diffrax.RESULTS.discrete_terminating_event_occurred also exists, and should usually also be treated as a success state, if you're using events in your code at all: (solution.result == diffrax.RESULTS.successful) | (solution.result == diffrax.RESULTS.discrete_terminating_event_occurred))

Which is basically the answer to my initial question in this issue, so I would close the issue.
